lib/gems/pending/util/MiqSshUtil.rb

```diff
@@ -149,18 +139,18 @@ def exec(cmd, doneStr = nil, stdin = nil)
 
         channel.on_close do |_channel|
           $log.debug "MiqSshUtil::exec - Command: #{cmd}, exit status: #{status}" if $log
-          unless signal.nil? || status.zero?
-            raise "MiqSshUtil::exec - Command #{cmd}, exited with signal #{signal}" unless signal.nil?
-            raise "MiqSshUtil::exec - Command #{cmd}, exited with status #{status}" if errBuf.empty?
-            raise "MiqSshUtil::exec - Command #{cmd} failed: #{errBuf}, status: #{status}"
+          if !signal.nil? || status.nonzero? || !errBuf.empty?
```

Comment:
I think the logic here is being changed.

Previously we would enter the block only `if signal != nil && status != 0`

Now we will enter the block if only `signal != nil`

I think the conditional you want here is:
```ruby
if !errBuf.empty? || (!signal.nil? && status.nonzero?)
```

Alternatively you could change a bit less and go with:
```ruby
if !(signal.nil? || status.zero?) || !errBuf.empty?
```

As a side note it seems like the content of the block is checking against the possibility that `signal` could still be nil even though it has to be non-nil in the original implementation. So if the conditional was wrong to begin with then this looks good. 

I'm just not sure which direction you were trying to go with this.

-----------------------------------------

FWIW this is why I don't like `unless` with more than one condition.

Reply:
@carbonin Not sure what you mean:

`if !signal.nil? || status.nonzero? || !errBuf.empty?`

We'll enter the block if any of those conditions is true.

Reply:
@djberg96 You applied the unless wrong, I think...

Previously...

```ruby
unless signal.nil? || status.zero?
# means
if !(signal.nil? || status.zero?)
# =>
if !signal.nil? && !status.zero?
# =>
if !signal.nil? && status.nonzero?
```

However you have (ignoring the new extra thing)

```ruby
if !signal.nil? || status.nonzero?
```

Reply:
@Fryguy right, I'm saying the logic I have now is the logic I want. If any of those conditions is met then something went wrong.

Reply:
@djberg96 I was just pointing out that the logic changed. You didn't call out this change in the description or commit message so I didn't know if it was intentional.

I know specs for this are probably tough to write, but having those would clarify intent quite nicely.